Variables, Data Types and Control Statements in C# (#1103)
section-io/engineering-education: articles/variables-data-types-and-control-statements-in-c#/index.md

```

Data types are used to classify the various types of variables and state the size of data the variable is supposed to store.

The different data types are:

### 1. Byte

The `byte` data type is used to store whole numbers from -128 to 127.

### 2. Int

The `int` (integer) data type is used to store whole numbers from -2,147,483,648 to 2,147,483,647.

### 3. Long

The `long` data type allows you to store whole numbers from -923,372,036,854,775,808 to 9,223,372,036,854,775,807.
We should add a letter **L** at the end of the long number to specify it is a long number.

### 4. Float

The `float` is used to store a fractional number and a decimal number of up to 6 decimal digits. We should add an **F** at the end of the decimal number to specify that it is a `float`.

### 5. Double

The `double` data type stores a fraction and a decimal number of up to 15 decimal digits.

*Note*: Double data type stores many decimal numbers than float. It is advisable to use double in most of the calculations.

### 6. Char

The `char` type is used to store **one** character.

### 7. String

The `string` data type is used to store **many** characters.

### 8. Bool

> The `bool` data type is used to store **true** or **false** value.

Reviewer: Hi @Jeff-mwangi 

The example here was good before. Please add them back but don't add the `Console.Writeline` in every example.
Also, instead of marking all the different types as headings, you can mark them as bullet points.

In this article, we are going to take a look at variables, data types, and control statements.

### Variables in C#

At some point, we need to utilize some data in our program, we can store it in a memory location and name the memory location as it gets simpler to get to it. We can achieve these by using variables.

Variables are names of a memory location for storing data types.

Variables are classified into various types:

- `int` is used to hold an integer(whole numbers) like 5,6,7,8.

- `char` is used to hold a character like 'A'.

- `string` is used to store text like "C# Language".

- `double` is used to store a fraction or a decimal value.

- `bool` is used to hold a true or false value.

### Syntax of declaring a variable

```C#
<data type> <variable_name> = value;
```

**Data type** is the C# type like `int, double, string, char or bool` while **variable_name** is the name of the variable (like x or name).
The **equal sign** initializes or provides the variable with a value.

**Rules of declaring a variable name:**

> - Keywords should not be used as a name for the variable.

Reviewer: @Jeff-mwangi This is not done yet.

console.WriteLine(c);
```

### 5. Double

The `double` data type stores a fraction and a decimal number of up to 15 decimal digits.

**Example**:

```C#
double d = 4.847376474;
console.WriteLine(d);
```
*Note*: Double data type stores many decimal numbers than float. It is advisable to use double in most of the calculations.

### 6. Char

The `char` type is used to store **one** character.

**Example**:

```C#
char letter='A';
console.WriteLine(letter);
```
### 7. String

The `string` data type is used to store **many** characters.

**Example**:
```C#
string name="My name is Geoffrey";
console.WriteLine(name);

```

### 8. Bool

The `bool` data type is used to store **true** or **false** value.

**Example**:
```C#
bool myBoolean=true;
bool nextBoolean=false;
console.WriteLine(myBoolean);//prints out true
console.WriteLine(nextBoolean);//prints out false
```

### Control Statements 
Control statements help to make decisions in a program, loop(iterate) through the conditions, select and change the flow of the conditions.

There are four types of control statements in C#:

- Decision-making statements
- Iteration statements
- Selection statements
- Jump statements

> We will look at decision-making statements and selection statements.

Reviewer: @Jeff-mwangi I noticed you removed the list. Please add it back and Provide the links to another article.

### 1. Decision-making statements

This type of condition test the conditions provided evaluating each of them either true or false.We use **if-else Statement** and **else if statement**.  

**Syntax of if-else Statement**:

```C#
if (condition)
{
  statements // block of code to be executed if the condition is true
}
else
 {
    statements // block of code to be executed if the condition is false
  }
```

If the statements in the `if condition` is **true**, the block of code inside the *if condition* is executed.

The `else statement` is executed only when the *if condition* is false.

**Example**:
 ```C#
int age = 18;
if (age < 15) 
   {
    Console.WriteLine("You are young.");
   } 
    else 
    {
     Console.WriteLine("You are old.");
    }
  //the program outputs "You are old"
```

> From the program above, we have set our age as 18. The program first checks for the first condition, that is if the age is less than 15. If it is true, the block of code in the *if condition* is executed otherwise the *else statement* is executed. In our program the if condition is false thus printing out the else statement.

Reviewer: ```suggestion
From the program above, we have set our age as 18. The program first checks for the condition, i.e., if the age is less than 15. If it is true, the *if* block is executed. Otherwise, the *else* block is executed. In our program, the condition is evaluated to *false* thus the *"You are old."* is printed.
```

### Else if Statement

> This condition is used if the first condition is **False**.

Reviewer: Better explanation required.

**Syntax of the Else if Statement**:

```C#
if (condition1)
  {
    statements // block of code to be executed if condition1 is true
  }
    else if (condition2)
      {
        statements // block of code to be executed if condition1 is false and condition2 is true
      }
    else
      {
        statements //block of code to be executed if both condition1 and condition2 is false
      }
```

**Example**:

```C#
int age = 18;
if (age < 10) 
  {
    Console.WriteLine("You are a child.");
  } 
    else if (age < 17) 
      {
        Console.WriteLine("You are a teenager.");
      } 
        else 
          {
            Console.WriteLine("You are old.");
          }
      //Prints out "You are old"
```

> From the program above, we have set our age as 18.Thus,our *if condition* is **false**. The program moves to the next condition,*else if condition* which is also **false**.

Reviewer: ```suggestion
From the program above, we have set our age as 18. Thus, our *if condition* is evaluated to **false**. The program moves to the next condition, *else if condition* which is also evaluated to **false**.
```

> Our program executes else statement since both if and else if conditions are **false**.

Reviewer: ```suggestion
Our program executes the else statement since both if and the else if conditions are **false**.
```

### 2. Selection Statement

> This type of condition statement is used to select a code block out of many. We use the switch-case statement in such a case.

Reviewer: better explanation required.

**Syntax of the switch-case Statement**:

```C#
switch(expression) 
  {
    case 1:
      // code block
      break;
    case 2:
      // code block
      break;
    default:
      // code block
      break;
  }
```

> The expression value can either be user input or program generated. When the expression matches with one of the cases, the code block is executed. The **break statement** is used to exit the testing of the cases when the match is found. The **default** condition is executed when any of the cases does not match with the expression on the cases.

Reviewer: ```suggestion
When the expression matches with one of the cases, that particular code block is executed. The **break** statement is used to exit from the code block. The **default** condition is executed when none of the cases match with the expression.
```
